### ISSUE TITLE
Headers are not applied for base interface methods

### ISSUE THREAD

**Issue by abobr** [reactiveui/refit#940]
**Describe the bug**

Headers from inheriting interface are expected to be applied to base methods

    public interface IBaseClient
    {
        [Get("/sum/")]
        Task<int> GetSum();
    }

    [Headers("X-User: ExtendedUser")]
    public interface IClientA: IBaseClient
    {
        [Get("/SomeEndpoint")] 
        Task SomeWork();
    }

IClientA.GetSum() should use [Headers("X-User: ExtendedUser")]

**Expected behavior**

Expected that IClientA methods (including inherited) will use headers
    X-User: ExtendedUser
as defined in IClientA

**Environment**

- OS:       Win Server 2016
- Refit: v5.0.23

**Comment by jamiehowarth0** [comment]
@abobr has this been fixed?

**Comment by abobr** [comment]
@jamiehowarth0 
no, but I looked through the code generating client and realized, that it's not clear what the priority of headers defined in inheriting class should be.
Will it override base interface's header only or base method's header too.
Documentation doesn't promise such an overriding to work, thus this is not a bug.
Feature request is a correct form.
From the logical perspective inheriting interface probably shouldn't interfere with how base interface's methods work.
To summarize, I'd introduced such overriding as a dynamically opt-in feature.
Maybe something like

```
public interface IBaseClient
{
    [Get("/sum/")]
    Task<int> GetSum();
}

[Override]     ///   or  [ImplicitNew]
[Headers("X-User: ExtendedUser")]
public interface IClientA: IBaseClient
{
    [Get("/SomeEndpoint")] 
    Task SomeWork();
}
```
